scapy/arch/linux.py

```diff
@@ -145,7 +147,7 @@ def attach_filter(s, bpf_filter, iface):
     nb = int(lines[0])
     bpf = ""
     for l in lines[1:]:
-        bpf += struct.pack("HBBI",*map(long,l.split()))
+        bpf += struct.pack("HBBI",*list(map(int,l.split())))
```

Comment:
You don't need `list()` here, this should be enough: `bpf += struct.pack("HBBI",*map(int, l.split()))`